### PR TITLE
chore(bench): add benches

### PR DESCRIPTION
**Pull request by lemarier**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [ ] No

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
WIP

Including `binary_size`, `cargo_deps`, `thread_count`, `max_memory`

### CODE REVIEW

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+name: benches
+
+on: [push, pull_request, workflow_dispatch]
```
Comment (wusyong) [inline .github/workflows/bench.yml]: Should we run this on every pull request? It should be okay to just run on pushing commits to dev branch imho.

File: .github/workflows/bench.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.BENCH_PAT }}
+          path: gh-pages
+          repository: tauri-apps/benchmark_results
+
+      - name: post benchmarks
+        if: |
+          github.repository == 'tauri-apps/wry' &&
+          (github.ref == 'refs/heads/dev' || github.ref == 'refs/heads/feat/benches')
```
Comment (wusyong) [inline .github/workflows/bench.yml]: This should just be `refs/heads/dev` after this PR is merged right?

File: bench/main.rs [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+  new_data.thread_count = thread_count;
+  new_data.syscall_count = syscall_count;
+
+  Ok(())
+}
+
+fn run_max_mem_benchmark() -> Result<HashMap<String, u64>> {
+  let mut results = HashMap::<String, u64>::new();
+
+  for (name, example_exe, return_code) in EXEC_TIME_BENCHMARKS {
```
Comment (wusyong) [inline bench/main.rs]: Is `return_code` still needed? I see all of them are set to None

File: .github/workflows/bench.yml (hunk 1 above)
Comment (lemarier) [reply .github/workflows/bench.yml]: Your right lets remove PR

File: .github/workflows/bench.yml (hunk 2 above)
Comment (lemarier) [reply .github/workflows/bench.yml]: Yes we can remove it

File: bench/main.rs (hunk 3 above)
Comment (lemarier) [reply bench/main.rs]: Not used in our current tests, we can remove it and re-add later if needed

Comment (wusyong) [reply bench/main.rs]: Yes let's do this. Or maybe comment out the part it supposes to do to remind us.

**Review by wusyong — APPROVED** [review]

**Pull request merged**